dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java

```diff
@@ -1607,7 +1620,43 @@ public boolean deleteAsset(String inode) throws Exception
         return result;
     }
 
-    public Map<String, Object> changeAssetMenuOrder (String inode, int newValue) throws ActionException, DotDataException {
+	/**
+	 * Verifies if a page is being used as a detail page for any content type
+	 * @param result
+	 * @param user
+	 * @param contentTypeAPI
+	 * @param id
+	 * @param cont
+	 * @return
+	 * @throws DotDataException
+	 * @throws LanguageException
+	 */
+	private Map<String, Object> validateRelatedContentType(Map<String, Object> result, User user,
+			ContentTypeAPI contentTypeAPI, Identifier id, Contentlet cont)
+			throws DotDataException, LanguageException {
+		List<ContentType> relatedContentTypes = contentTypeAPI.search("page_detail='" + id.getId() + "'");
+
+		HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+		String uri = htmlPageAssetAPI.fromContentlet(cont).getURI();
+		//Verifies if the page is related to any content type
+		if (relatedContentTypes != null && !relatedContentTypes.isEmpty()){
+            StringBuilder relatedPagesMessage = new StringBuilder();
+            relatedPagesMessage.append(UtilMethods.escapeSingleQuotes(LanguageUtil.get(user,
+                    "HTML-Page-related-content-type-delete-error")));
+
+            relatedPagesMessage.append(relatedContentTypes.stream()
+                    .map(t -> t.name() + " - Detail Page: " + uri)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 't' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)